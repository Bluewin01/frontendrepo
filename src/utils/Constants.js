export const theme = {
  color: {
    w: "#FFFFFF",
    k: "#000000",
    p1: "#D31145",
    p2: "#596C80",
    p3: "#554344",
    p4: "#363E3F",
    p5: "#e6edeb",
    s1: "#E86487",
    s2: "#FBBF56",
    s3: "#97CB5D",
    s4: "#46D3BD",
    s5: "#22A8DA",
    s6: "#9962D2",
    t1: "#F6CCD7",
    t2: "#B8123E",
    t3: "#F9E1B8",
    t4: "#C36418",
    t5: "#C6EB9E",
    t6: "#65962E",
    b1: "#D4EDF1",
    b2: "#FAF9F3",
    b3: "#F0EEE1",
    b4: "#DBDAD4",
    b5: "#B3B1A8",
    b6: "#6F6E68",
    b8: "#302F2C",
    b9: "#292025",
    u2: "#FF5500",
    u3: "#405B70",
    u4: "#769FAB",
  },
};

export const BASE_API = `${process.env.REACT_APP_BASE_API_URL}`;
