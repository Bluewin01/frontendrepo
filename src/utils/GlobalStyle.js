import { createGlobalStyle } from 'styled-components';
import 'antd/dist/antd.css';
import '@aia-digital/ui-library/lib/static/styles.css';

import AIABoldEOT from '../assets/fonts/AIAEverest-Bold.eot';
import AIABoldWOFF from '../assets/fonts/AIAEverest-Bold.woff';
import AIABoldWOFF2 from '../assets/fonts/AIAEverest-Bold.woff2';
import AIABoldTTF from '../assets/fonts/AIAEverest-Bold.ttf';
import AIABoldSVG from '../assets/fonts/AIAEverest-Bold.svg';

import AIACondensedEOT from '../assets/fonts/AIAEverest-Condensed.eot';
import AIACondensedWOFF from '../assets/fonts/AIAEverest-Condensed.woff';
import AIACondensedWOFF2 from '../assets/fonts/AIAEverest-Condensed.woff2';
import AIACondensedTTF from '../assets/fonts/AIAEverest-Condensed.ttf';
import AIACondensedSVG from '../assets/fonts/AIAEverest-Condensed.svg';

import AIACondensedMediumEOT from '../assets/fonts/AIAEverest-CondensedMedium.eot';
import AIACondensedMediumWOFF from '../assets/fonts/AIAEverest-CondensedMedium.woff';
import AIACondensedMediumWOFF2 from '../assets/fonts/AIAEverest-CondensedMedium.woff2';
import AIACondensedMediumTTF from '../assets/fonts/AIAEverest-CondensedMedium.ttf';
import AIACondensedMediumSVG from '../assets/fonts/AIAEverest-CondensedMedium.svg';

import AIAExtraBoldEOT from '../assets/fonts/AIAEverest-ExtraBold.eot';
import AIAExtraBoldWOFF from '../assets/fonts/AIAEverest-ExtraBold.woff';
import AIAExtraBoldWOFF2 from '../assets/fonts/AIAEverest-ExtraBold.woff2';
import AIAExtraBoldTTF from '../assets/fonts/AIAEverest-ExtraBold.ttf';
import AIAExtraBoldSVG from '../assets/fonts/AIAEverest-ExtraBold.svg';

import AIAMediumEOT from '../assets/fonts/AIAEverest-Medium.eot';
import AIAMediumWOFF from '../assets/fonts/AIAEverest-Medium.woff';
import AIAMediumWOFF2 from '../assets/fonts/AIAEverest-Medium.woff2';
import AIAMediumTTF from '../assets/fonts/AIAEverest-Medium.ttf';
import AIAMediumSVG from '../assets/fonts/AIAEverest-Medium.svg';

import AIARegularEOT from '../assets/fonts/AIAEverest-Regular.eot';
import AIARegularWOFF from '../assets/fonts/AIAEverest-Regular.woff';
import AIARegularWOFF2 from '../assets/fonts/AIAEverest-Regular.woff2';
import AIARegularTTF from '../assets/fonts/AIAEverest-Regular.ttf';
import AIARegularSVG from '../assets/fonts/AIAEverest-Regular.svg';

const GlobalStyle = createGlobalStyle`
  @font-face {
    font-family: 'AIABold';
    src: url('${AIABoldEOT}');
    src: url('${AIABoldEOT}?#iefix') format('embedded-opentype'),
        url('${AIABoldWOFF}') format('woff'),
        url('${AIABoldWOFF2}') format('woff2'),
        url('${AIABoldTTF}') format('truetype'),
        url('${AIABoldSVG}#AIAEverestBold') format('svg');
    font-weight: normal;
    font-style: normal;
  }
  @font-face {
    font-family: 'AIACondensed';
    src: url('${AIACondensedEOT}');
    src: url('${AIACondensedEOT}?#iefix') format('embedded-opentype'),
        url('${AIACondensedWOFF}') format('woff'),
        url('${AIACondensedWOFF2}') format('woff2'),
        url('${AIACondensedTTF}') format('truetype'),
        url('${AIACondensedSVG}#AIAEverestCondensed') format('svg');
    font-weight: normal;
    font-style: normal;
  }
  @font-face {
    font-family: 'AIACondensedMedium';
    src: url('${AIACondensedMediumEOT}');
    src: url('${AIACondensedMediumEOT}?#iefix') format('embedded-opentype'),
        url('${AIACondensedMediumWOFF}') format('woff'),
        url('${AIACondensedMediumWOFF2}') format('woff2'),
        url('${AIACondensedMediumTTF}') format('truetype'),
        url('${AIACondensedMediumSVG}#AIAEverestCondensedMedium') format('svg');
    font-weight: 500;
    font-style: normal;
  }
  @font-face {
    font-family: 'AIAExtraBold';
    src: url('${AIAExtraBoldEOT}');
    src: url('${AIAExtraBoldEOT}?#iefix') format('embedded-opentype'),
        url('${AIAExtraBoldWOFF}') format('woff'),
        url('${AIAExtraBoldWOFF2}') format('woff2'),
        url('${AIAExtraBoldTTF}') format('truetype'),
        url('${AIAExtraBoldSVG}#AIAEverestExtraBold') format('svg');
    font-weight: normal;
    font-style: normal;
  }
  @font-face {
    font-family: 'AIAMedium';
    src: url('${AIAMediumEOT}');
    src: url('${AIAMediumEOT}?#iefix') format('embedded-opentype'),
        url('${AIAMediumWOFF}') format('woff'),
        url('${AIAMediumWOFF2}') format('woff2'),
        url('${AIAMediumTTF}') format('truetype'),
        url('${AIAMediumSVG}#AIAEverestMedium') format('svg');
    font-weight: 600;
    font-style: normal;
  }
  @font-face {
    font-family: 'AIARegular';
    src: url('${AIARegularEOT}');
    src: url('${AIARegularEOT}?#iefix') format('embedded-opentype'),
        url('${AIARegularWOFF}') format('woff'),
        url('${AIARegularWOFF2}') format('woff2'),
        url('${AIARegularTTF}') format('truetype'),
        url('${AIARegularSVG}#AIAEverestRegular') format('svg');
    font-weight: 400;
    font-style: normal;
  }

  body h1 {
    font-family: "AIACondensedMedium", "Calibri", sans-serif;
    font-size: 96px;
    line-height: 84px;
    font-weight: normal;
    margin-top: 0px;
    margin-bottom: 0px;
    color: ${({ theme }) => theme.color.p3}
  }

  body h2 {
    font-family: "AIACondensedMedium", "Calibri", sans-serif;
    font-size: 72px;
    line-height: 64px;
    font-weight: normal;
    margin-top: 0px;
    margin-bottom: 0px;
    color: ${({ theme }) => theme.color.p3}
  }

  body h3 {
    font-family: "AIACondensedMedium", "Calibri", sans-serif;
    font-size: 48px;
    line-height: 44px;
    font-weight: normal;
    margin-top: 0px;
    margin-bottom: 0px;
    color: ${({ theme }) => theme.color.p3}
  }

  body h4 {
    font-family: "AIACondensedMedium", "Calibri", sans-serif;
    font-size: 32px;
    line-height: 28px;
    font-weight: normal;
    margin-top: 0px;
    margin-bottom: 0px;
    color: ${({ theme }) => theme.color.p3}
  }

  body h5 {
    font-family: "AIACondensedMedium", "Calibri", sans-serif;
    font-size: 24px;
    line-height: 22px;
    font-weight: normal;
    margin-top: 0px;
    margin-bottom: 0px;
    color: ${({ theme }) => theme.color.p3}
  }

  body h6 {
    font-family: "AIACondensedMedium", "Calibri", sans-serif;
    font-size: 20px;
    line-height: 20px;
    font-weight: normal;
    margin-top: 0px;
    margin-bottom: 0px;
    color: ${({ theme }) => theme.color.p3}
  }

  body p, ul, ol {
    font-family: "AIARegular", Arial, "Helvetica Neue", Helvetica, sans-serif;
    font-size: 15px;
    line-height: 20px;
    margin-bottom: 0px;
    color: ${({ theme }) => theme.color.p3}
  }
`;

export default GlobalStyle;
