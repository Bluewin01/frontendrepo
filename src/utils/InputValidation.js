export const validateIsInputEmpty = (props) => {
  const inputValues = Object.values(props);
  const inputKeys = Object.keys(props);
  let invalidInputList = [];

  inputValues.forEach((input, index) => {
    if (
      input === "" ||
      input === undefined ||
      input === null ||
      input.match(/^ *$/) !== null
    ) {
      invalidInputList.push(index);
    }
  });

  let validationResult = [];
  invalidInputList.forEach((fieldIndex) => {
    validationResult.push(inputKeys[fieldIndex]);
  });

  return validationResult;
};

export const validateIsInputListEmpty = (props) => {
  let invalidInputList = [];

  props.forEach((input) => {
    if (
      input.value === "" ||
      input.value === undefined ||
      input.value === null ||
      input.value.match(/^ *$/) !== null
    ) {
      invalidInputList.push(input.name);
    }
  });

  return invalidInputList;
};

export const validateEmailFormat = (props) => {
  const inputValues = Object.values(props);
  const inputKeys = Object.keys(props);
  let invalidEmailList = [];

  inputValues.forEach((email, index) => {
    if (
      !email.match(
        /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
      )
    ) {
      invalidEmailList.push(index);
    }
  });

  let validationResult = [];
  invalidEmailList.forEach((fieldIndex) => {
    validationResult.push(inputKeys[fieldIndex]);
  });

  return validationResult;
};

export const validateEmailListFormat = (props) => {
  let invalidEmailList = [];

  props.forEach((email) => {
    if (email.value !== "") {
      if (
        !email.value.match(
          /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
        )
      ) {
        invalidEmailList.push(email.name);
      }
    }
  });

  return invalidEmailList;
};
