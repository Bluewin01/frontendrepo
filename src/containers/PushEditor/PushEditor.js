import "../../assets/css/editor.css";
import React, { useEffect, useState } from "react";
import { Form, Container, Button } from "react-bootstrap";
import History from "../../routes/History";
import { connect } from "react-redux";
import * as TemplateActions from "../../store/Template/TemplateActions";
import PropTypes from "prop-types";
import { ToastContainer, toast } from "react-toastify";
import "react-toastify/dist/ReactToastify.css";
import Loader from "../../components/Loader/Loader";
import { createStructuredSelector } from "reselect";
import {
  makeSelectTemplateObject,
  makeSelectError,
  makeSelectResponse,
} from "../../store/Template/TemplateReselect";
import { validateIsInputEmpty } from "../../utils/InputValidation";
import { ErrorMessage } from "../../components/ErrorMessage/ErrorMessage";
import { Modal } from "../../components/Modal/Modal";

const PushEditor = (props) => {
  const {
    initiateSaveTemplate,
    templateObject,
    initiateEditTemplate,
    error,
    response,
    initiateSetErrorToDefault,
    initiateSetResponseToDefault,
  } = props;

  const [loading, showLoader] = useState(false);
  const [name, setName] = useState("");
  const [subject, setSubject] = useState("");
  const [description, setDescription] = useState("");
  const [value, setValue] = useState("");
  const [nameError, setNameError] = useState(false);
  const [subjectError, setSubjectError] = useState("");
  const [descriptionError, setDescriptionError] = useState("");
  const [valueError, setValueError] = useState("");
  const [showSaveModal, setShowSaveModal] = useState(false);
  const [showCancelModal, setShowCancelModal] = useState(false);

  const initiateErrorMessage = () => {
    setNameError("");
    setSubjectError("");
    setDescriptionError("");
    setValueError("");
  };

  const buttons = [
    {
      name: "Save",
      color: "danger",
      testId: "showSaveModal",
      handler: () => {
        setShowSaveModal(true);
      },
    },
    {
      name: "Cancel",
      color: "light",
      testId: "showCancelModal",
      handler: () => {
        setShowCancelModal(true);
      },
    },
  ];

  useEffect(() => {
    if (templateObject !== null) {
      const { name, subject, description, value } = templateObject;
      const { value: obj } = value;
      setName(name);
      setSubject(subject);
      setDescription(description);
      setValue(obj);
    } else {
      setName("");
      setSubject("");
      setDescription("");
      setValue("");
    }
  }, [templateObject]);

  const handleSave = () => {
    initiateErrorMessage();
    const type = "Push";
    const validationResult = validateIsInputEmpty({
      name,
      subject,
      description,
      value,
    });
    if (validationResult.length === 0) {
      if (templateObject !== null) {
        initiateEditTemplate(name, subject, description, value, type);
      } else {
        initiateSaveTemplate(name, subject, description, value, type);
      }
      showLoader(true);
      setTimeout(() => {
        showLoader(false);
        History.push("/home");
      }, 1000);
    } else {
      if (validationResult.includes("name"))
        setNameError("Template Name is required.");
      if (validationResult.includes("subject"))
        setSubjectError("Template Title is required.");
      if (validationResult.includes("description"))
        setDescriptionError("Subtitle is required.");
      if (validationResult.includes("value"))
        setValueError("Message is required.");
    }
    setShowSaveModal(false);
  };

  const handleCancel = () => {
    History.push("/home");
  };

  useEffect(() => {
    if (error) toast.error(error);
    if (response) toast.success(response);
    initiateSetErrorToDefault();
    initiateSetResponseToDefault();
  }, [error, response]);

  return (
    <Container>
      <ToastContainer
        position="top-right"
        autoClose={5000}
        hideProgressBar
        newestOnTop={false}
        closeOnClick
        rtl={false}
        pauseOnFocusLoss
        draggable
        pauseOnHover={false}
      />
      {loading ? <Loader></Loader> : null}
      <Form>
        <br></br>
        <h4 className="text-center mb-5">Push Template Editor</h4>
        <div className="row">
          <Form.Group className="col mb-6" controlId="template_name">
            <Form.Label className="required">Template Name</Form.Label>
            <Form.Control
              type="text"
              name="name"
              data-testid="name"
              value={name}
              onChange={(e) => {
                setName(e.target.value);
                if (validateIsInputEmpty(e.target.value).length === 0)
                  setNameError("");
              }}
              autoComplete="off"
            />
            <ErrorMessage error={nameError} />
          </Form.Group>
          <Form.Group className="col mb-6" controlId="message_sub">
            <Form.Label className="required">Template Title</Form.Label>
            <Form.Control
              type="text"
              name="subject"
              data-testid="subject"
              value={subject}
              onChange={(e) => {
                setSubject(e.target.value);
                if (validateIsInputEmpty(e.target.value).length === 0)
                  setSubjectError("");
              }}
              autoComplete="off"
            />
            <ErrorMessage error={subjectError} />
          </Form.Group>
          <Form.Group className="col mb-6" controlId="template_name">
            <Form.Label className="required">Subtitle</Form.Label>
            <Form.Control
              type="text"
              name="description"
              data-testid="description"
              value={description}
              onChange={(e) => {
                setDescription(e.target.value);
                if (validateIsInputEmpty(e.target.value).length === 0)
                  setDescriptionError("");
              }}
              autoComplete="off"
            />
            <ErrorMessage error={descriptionError} />
          </Form.Group>
        </div>
        <Form.Group className="mb-3" controlId="message">
          <Form.Label className="required">Message</Form.Label>
          <Form.Control
            as="textarea"
            placeholder="Sample Notification"
            maxLength="30"
            rows={3}
            data-testid="message"
            value={value}
            onChange={(e) => {
              setValue(e.target.value);
              if (validateIsInputEmpty(e.target.value).length === 0)
                setValueError("");
            }}
            autoComplete="off"
          />
          <ErrorMessage error={valueError} />
        </Form.Group>
        <div className="text-right mt-3">
          {buttons.map((button) => {
            return (
              <Button
                data-testid={button.testId}
                key={button.name}
                className="Button"
                variant={button.color}
                onClick={button.handler}
              >
                {button.name}
              </Button>
            );
          })}
        </div>
      </Form>
      {showSaveModal && (
        <Modal
          headerTitle={`Are you sure you want to save Push template`}
          buttons={[
            {
              name: "Confirm",
              color: "danger",
              testId: "triggerHandler",
              handler: handleSave,
            },
            {
              name: "Cancel",
              color: "light",
              testId: "closeModal",
              handler: () => setShowSaveModal(false),
            },
          ]}
        />
      )}
      {showCancelModal && (
        <Modal
          headerTitle={`All changes will be lost. Are you sure to continue?`}
          buttons={[
            {
              name: "Confirm",
              color: "danger",
              handler: handleCancel,
            },
            {
              name: "Cancel",
              color: "light",
              handler: () => setShowCancelModal(false),
            },
          ]}
        />
      )}
    </Container>
  );
};

const mapStateToProps = createStructuredSelector({
  templateObject: makeSelectTemplateObject(),
  error: makeSelectError(),
  response: makeSelectResponse(),
});

const mapDispatchToProps = (dispatch) => {
  return {
    initiateSaveTemplate: (name, subject, description, value, type) =>
      dispatch(
        TemplateActions.saveTemplateRequest({
          name,
          subject,
          description,
          value,
          type,
        })
      ),
    initiateEditTemplate: (name, subject, description, value, type) =>
      dispatch(
        TemplateActions.editTemplateRequest({
          name,
          subject,
          description,
          value,
          type,
        })
      ),
    initiateSetResponseToDefault: () =>
      dispatch(TemplateActions.setResponseToDefault()),
    initiateSetErrorToDefault: () =>
      dispatch(TemplateActions.setErrorToDefault()),
  };
};

PushEditor.propTypes = {
  initiateSaveTemplate: PropTypes.func,
  initiateEditTemplate: PropTypes.func,
  templateObject: PropTypes.object,
  response: PropTypes.string,
  error: PropTypes.string,
  initiateSetResponseToDefault: PropTypes.func,
  initiateSetErrorToDefault: PropTypes.func,
};

export default connect(mapStateToProps, mapDispatchToProps)(PushEditor);
