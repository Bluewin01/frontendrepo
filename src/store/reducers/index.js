import { combineReducers } from "redux";
import { configureStore } from "@reduxjs/toolkit";
import lang from "../Lang/LangReducer";
import loader from "../Loader/LoaderReducer";
import Auth from "../Auth/AuthReducer";
import Template from "../Template/TemplateReducer";

const rootReducer = combineReducers({
  lang,
  loader,
  Auth,
  Template,
});

export const setupStore = (preloadedState) => {
  return configureStore({
    reducer: rootReducer,
    preloadedState,
  });
};

export default rootReducer;
