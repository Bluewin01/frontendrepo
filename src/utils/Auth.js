// Service to check authentication for user and to signOut
import * as AuthActions from "../store/Auth/AuthActions";
import store from "../store";

const Auth = {
  signOut() {
    store.dispatch(AuthActions.signOut);
  },
  isAuth(jwt) {
    if (!jwt) {
      return false;
    } else {
      return true;
    }
  },
};
export default Auth;
