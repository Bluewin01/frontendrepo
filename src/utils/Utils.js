// test-utils.jsx
import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
import { setupStore } from "../store/reducers";

export function render(
  ui,
  { preloadedState, store = setupStore(preloadedState), ...renderOptions } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }

  // Return an object with the store and all of RTL's query functions
  return { store, ...rtlRender(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export * from "@testing-library/react";
