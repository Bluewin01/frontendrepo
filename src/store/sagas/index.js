import { AuthSaga } from "../Auth/AuthSaga";
import { TemplateSaga } from "../Template/TemplateSaga";
import { fork, all } from "redux-saga/effects";

export function* watchSagas() {
  yield all([AuthSaga(), TemplateSaga()]);
}
