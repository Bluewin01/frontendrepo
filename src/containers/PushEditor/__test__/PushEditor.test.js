import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { screen, cleanup, fireEvent, act } from "@testing-library/react";
import { render } from "../../../utils/Utils";
import renderer from "react-test-renderer";
import store from "../../../store";
import { Provider } from "react-redux";
import PushEditor from "../PushEditor";

jest.useFakeTimers();

describe("Containers: Push Editor", () => {
  beforeAll(() => {
    Object.defineProperty(window, "matchMedia", {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // Deprecated
        removeListener: jest.fn(), // Deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });
  afterEach(cleanup);
  describe("Positive Case", () => {
    it("should be render correctly", () => {
      const tree = renderer
        .create(
          <Provider store={store}>
            <PushEditor />
          </Provider>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it("should be able to render email page with error", () => {
      render(<PushEditor />, {
        preloadedState: {
          Template: {
            error: "fakeError",
            response: null,
            templateObject: null,
          },
        },
      });
    });

    it("should be able to render email page with response", () => {
      render(<PushEditor />, {
        preloadedState: {
          Template: {
            error: null,
            response: "fakeResponse",
            templateObject: null,
          },
        },
      });
    });

    it("should can input all the fields", () => {
      render(<PushEditor />);
      const mockTemplateName = "Mock Template Name";
      const mockSubject = "Mock Template Subject";
      const mockDescription = "Mock Template Description";
      const mockMessage = "Mock Template Message";

      const inputName = screen.getByTestId("name");
      const inputSubject = screen.getByTestId("subject");
      const inputDescription = screen.getByTestId("description");
      const inputMessage = screen.getByTestId("message");

      fireEvent.change(inputName, { target: { value: mockTemplateName } });
      fireEvent.change(inputSubject, { target: { value: mockSubject } });
      fireEvent.change(inputDescription, {
        target: { value: mockDescription },
      });
      fireEvent.change(inputMessage, { target: { value: mockMessage } });

      expect(inputName.value).toBe(mockTemplateName);
      expect(inputSubject.value).toBe(mockSubject);
      expect(inputDescription.value).toBe(mockDescription);
      expect(inputMessage.value).toBe(mockMessage);
    });

    it("should can be save the template after input all fields", async () => {
      render(<PushEditor />);
      const mockTemplateName = "Mock Template Name";
      const mockSubject = "Mock Template Subject";
      const mockDescription = "Mock Template Description";
      const mockMessage = "Mock Template Message";

      const inputName = screen.getByTestId("name");
      const inputSubject = screen.getByTestId("subject");
      const inputDescription = screen.getByTestId("description");
      const inputMessage = screen.getByTestId("message");

      fireEvent.change(inputName, { target: { value: mockTemplateName } });
      fireEvent.change(inputSubject, { target: { value: mockSubject } });
      fireEvent.change(inputDescription, {
        target: { value: mockDescription },
      });
      fireEvent.change(inputMessage, { target: { value: mockMessage } });

      const saveButton = screen.getByTestId("showSaveModal");
      fireEvent.click(saveButton);

      const confirmButton = screen.getByTestId("triggerHandler");
      fireEvent.click(confirmButton);
      act(() => jest.runOnlyPendingTimers());

      expect(saveButton).toHaveTextContent("Save");
      expect(confirmButton).toHaveTextContent("Confirm");
    });

    it("should can be send the template after input all fields", async () => {
      render(<PushEditor />);
      const mockTemplateName = "Mock Template Name";
      const mockSubject = "Mock Template Subject";
      const mockDescription = "Mock Template Description";
      const mockMessage = "Mock Template Message";

      const inputName = screen.getByTestId("name");
      const inputSubject = screen.getByTestId("subject");
      const inputDescription = screen.getByTestId("description");
      const inputMessage = screen.getByTestId("message");

      fireEvent.change(inputName, { target: { value: mockTemplateName } });
      fireEvent.change(inputSubject, { target: { value: mockSubject } });
      fireEvent.change(inputDescription, {
        target: { value: mockDescription },
      });
      fireEvent.change(inputMessage, { target: { value: mockMessage } });

      const sendButton = screen.getAllByRole("button")[0];
      fireEvent.click(sendButton);

      const confirmButton = screen.getAllByRole("button")[0];
      fireEvent.click(confirmButton);

      expect(confirmButton).toHaveTextContent("Confirm");
    });

    it("should be redirect to /home route when click cancel button", () => {
      render(<PushEditor />);
      const cancelButton = screen.getByText("Cancel");
      fireEvent.click(cancelButton);

      expect(
        screen.getByText("All changes will be lost. Are you sure to continue?")
      ).toBeInTheDocument();

      const confirmButton = screen.getByText("Confirm");
      fireEvent.click(confirmButton);
      expect(global.window.location.href).toContain("/home");
    });

    it("should be able to save with template object and redirect to home page", async () => {
      render(<PushEditor />, {
        preloadedState: {
          Template: {
            error: null,
            response: null,
            templateObject: {
              name: "fakeName",
              subject: "fakeSubject",
              description: "fakeDescription",
              value: { value: "fakeMessage" },
            },
          },
        },
      });

      const saveButton = screen.getByTestId("showSaveModal");
      fireEvent.click(saveButton);

      const confirmButton = screen.getByTestId("triggerHandler");
      fireEvent.click(confirmButton);

      expect(global.window.location.href).toContain("/home");
    });
  });

  describe("Negative Case", () => {
    it("should show error message for template name when saving with no template name", () => {
      render(<PushEditor />);
      const mockSubject = "Mock Template Subject";
      const mockDescription = "Mock Template Description";
      const mockMessage = "Mock Template Message";
      const mockErrorMessage = "Template Name is required.";

      const inputSubject = screen.getByTestId("subject");
      const inputDescription = screen.getByTestId("description");
      const inputMessage = screen.getByTestId("message");

      fireEvent.change(inputSubject, { target: { value: mockSubject } });
      fireEvent.change(inputDescription, {
        target: { value: mockDescription },
      });
      fireEvent.change(inputMessage, { target: { value: mockMessage } });

      const saveButton = screen.getByTestId("showSaveModal");
      fireEvent.click(saveButton);

      const confirmButton = screen.getByTestId("triggerHandler");
      fireEvent.click(confirmButton);

      const errorMessage = screen.getByTestId("errorMessage");
      expect(errorMessage).toHaveTextContent(mockErrorMessage);
    });

    it("should show error message for template subject when saving with no template subject", () => {
      render(<PushEditor />);
      const mockTemplateName = "Mock Template Name";
      const mockDescription = "Mock Template Description";
      const mockMessage = "Mock Template Message";
      const mockErrorMessage = "Template Title is required.";

      const inputName = screen.getByTestId("name");
      const inputDescription = screen.getByTestId("description");
      const inputMessage = screen.getByTestId("message");

      fireEvent.change(inputName, { target: { value: mockTemplateName } });
      fireEvent.change(inputDescription, {
        target: { value: mockDescription },
      });
      fireEvent.change(inputMessage, { target: { value: mockMessage } });

      const saveButton = screen.getByTestId("showSaveModal");
      fireEvent.click(saveButton);

      const confirmButton = screen.getByTestId("triggerHandler");
      fireEvent.click(confirmButton);

      const errorMessage = screen.getByTestId("errorMessage");
      expect(errorMessage).toHaveTextContent(mockErrorMessage);
    });

    it("should show error message for template subject when saving with no template subtitle", () => {
      render(<PushEditor />);
      const mockTemplateName = "Mock Template Name";
      const mockSubject = "Mock Template Subject";
      const mockMessage = "Mock Template Message";
      const mockErrorMessage = "Subtitle is required.";

      const inputName = screen.getByTestId("name");
      const inputSubject = screen.getByTestId("subject");
      const inputMessage = screen.getByTestId("message");

      fireEvent.change(inputName, { target: { value: mockTemplateName } });
      fireEvent.change(inputSubject, { target: { value: mockSubject } });
      fireEvent.change(inputMessage, {
        target: { value: inputMessage },
      });

      const saveButton = screen.getByTestId("showSaveModal");
      fireEvent.click(saveButton);

      const confirmButton = screen.getByTestId("triggerHandler");
      fireEvent.click(confirmButton);

      const errorMessage = screen.getByTestId("errorMessage");
      expect(errorMessage).toHaveTextContent(mockErrorMessage);
    });

    it("should show error message for template subject when saving with no template message", () => {
      render(<PushEditor />);
      const mockTemplateName = "Mock Template Name";
      const mockSubject = "Mock Template Subject";
      const mockDescription = "Mock Template Description";
      const mockErrorMessage = "Message is required.";

      const inputName = screen.getByTestId("name");
      const inputSubject = screen.getByTestId("subject");
      const inputDescription = screen.getByTestId("description");

      fireEvent.change(inputName, { target: { value: mockTemplateName } });
      fireEvent.change(inputSubject, { target: { value: mockSubject } });
      fireEvent.change(inputDescription, {
        target: { value: mockDescription },
      });

      const saveButton = screen.getByTestId("showSaveModal");
      fireEvent.click(saveButton);

      const confirmButton = screen.getByTestId("triggerHandler");
      fireEvent.click(confirmButton);

      const errorMessage = screen.getByTestId("errorMessage");
      expect(errorMessage).toHaveTextContent(mockErrorMessage);
    });
  });
});
