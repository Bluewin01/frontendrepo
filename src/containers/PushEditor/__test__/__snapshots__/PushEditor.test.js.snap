// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Containers: Push Editor Positive Case should be render correctly 1`] = `
<div
  className="container"
>
  <div
    className="Toastify"
  />
  <form
    className=""
  >
    <br />
    <h4
      className="text-center mb-5"
    >
      Push Template Editor
    </h4>
    <div
      className="row"
    >
      <div
        className="col mb-6 form-group"
      >
        <label
          className="required form-label"
          htmlFor="template_name"
        >
          Template Name
        </label>
        <input
          autoComplete="off"
          className="form-control"
          data-testid="name"
          id="template_name"
          name="name"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="col mb-6 form-group"
      >
        <label
          className="required form-label"
          htmlFor="message_sub"
        >
          Template Title
        </label>
        <input
          autoComplete="off"
          className="form-control"
          data-testid="subject"
          id="message_sub"
          name="subject"
          onChange={[Function]}
          type="text"
          value=""
        />
        
      </div>
      <div
        className="col mb-6 form-group"
      >
        <label
          className="required form-label"
          htmlFor="template_name"
        >
          Subtitle
        </label>
        <input
          autoComplete="off"
          className="form-control"
          data-testid="description"
          id="template_name"
          name="description"
          onChange={[Function]}
          type="text"
          value=""
        />
        
      </div>
    </div>
    <div
      className="mb-3 form-group"
    >
      <label
        className="required form-label"
        htmlFor="message"
      >
        Message
      </label>
      <textarea
        autoComplete="off"
        className="form-control"
        data-testid="message"
        id="message"
        maxLength="30"
        onChange={[Function]}
        placeholder="Sample Notification"
        rows={3}
        value=""
      />
      
    </div>
    <div
      className="text-right mt-3"
    >
      <button
        className="Button btn btn-danger"
        data-testid="showSaveModal"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Save
      </button>
      <button
        className="Button btn btn-light"
        data-testid="showCancelModal"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Cancel
      </button>
    </div>
  </form>
</div>
`;
